01-node-tutorial/answers/customEmitter.js

```diff
@@ -0,0 +1,39 @@
+const EventEmitter = require("events");
+
+const emitter = new EventEmitter();
+
+const poems = [
+    { pink: "What is pink? a rose is pink\nBy a fountain's brink.\n" },
+    { red: "What is red? a poppy's red\nIn its barley bed.\n" },
+    { blue: "What is blue? the sky is blue\nWhere the clouds float thro'.\n" },
+    { white: "What is white? a swan is white\nSailing in the light.\n" },
+    {
+        yellow: "What is yellow? pears are yellow,\nRich and ripe and mellow.\n",
+    },
+    {
+        green: "What is green? the grass is green,\nWith small flowers between.\n",
+    },
+    {
+        vilolet:
+            "What is violet? clouds are violet\nIn the summer twilight.\n",
+    },
+    { orange: "What is orange? Why, an orange,\nJust an orange!\n" },
+];
+
+const colors = poems.reduce((acc, obj) => {
```

Comment:
Just a heads up, you can do this using the `flatMap` pattern as well.

This will take the keys and [flatten](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) the array of arrays into a single array.

```javascript
const colors = poems.map((p) => Object.keys(p)).flat(1);
```
Taking it one step further, there's a [convenience method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) for this.
```javascript
const colors = poems.flatMap((p) => Object.keys(p));
```

No need to change it though, what you have here is completely valid.